core/src/test/scala/org/ensime/config/EnsimeConfigSpec.scala

```diff
@@ -79,33 +78,28 @@ class EnsimeConfigSpec extends EnsimeSpec {
     })
   }
 
-  it should "base class paths on source-mode value" in {
-    List(true, false) foreach { (sourceMode: Boolean) =>
-      withTempDir { dir =>
-        val abc = dir / "abc"
-        val cache = dir / ".ensime_cache"
-        val javaHome = File(Properties.javaHome)
+  it should "base class paths on ensime.sourceMode value" in withTempDir { dir =>
+    val abc = dir / "abc"
+    val cache = dir / ".ensime_cache"
+    val javaHome = File(Properties.javaHome)
 
-        abc.mkdirs()
-        cache.mkdirs()
+    abc.mkdirs()
+    cache.mkdirs()
 
-        test(dir, s"""
+    test(dir, s"""
 (:name "project"
  :scala-version "2.10.4"
  :java-home "$javaHome"
  :root-dir "$dir"
  :cache-dir "$cache"
- :source-mode ${if (sourceMode) "t" else "nil"}
+ :source-mode ${if (Properties.propOrFalse("ensime.sourceMode")) "t" else "nil"}
  :subprojects ((:name "module1"
                 :scala-version "2.10.4"
                 :targets ("$abc"))))""", { implicit config =>
-          config.sourceMode shouldBe sourceMode
-          config.compileClasspath shouldBe (
-            if (sourceMode) Set.empty else Set(abc)
-          )
-        })
-      }
-    }
+      config.compileClasspath shouldBe (
+        if (Properties.propOrFalse("ensime.sourceMode")) Set.empty else Set(abc)
```

Comment:
In that case, there is nothing this test would cover that's not already covered and can be deleted.